File: final_dataset_A/generated_code/chatgpt/python/receive_process_webhook_detailed-chatgpt.py
from flask import Flask, request, jsonify
from datetime import datetime

app = Flask(__name__)

processed_events = []

def process_event(event):
    event_type = event.get("type", "unknown")

    return {
        "type": event_type,
        "payload": event,
        "processed_at": datetime.now().isoformat()
    }

@app.route("/webhook", methods=["POST"])
def receive_webhook():
    # Parse incoming JSON payload
    event = request.get_json()

    if not event:
        return jsonify({
            "success": False,
            "message": "Invalid JSON payload."
        }), 400

    # Process the event
    processed_event = process_event(event)

    # Record the processed event
    processed_events.append(processed_event)

    print(f"Processed event: {processed_event['type']}")

    # Return confirmation response
    return jsonify({
        "success": True,
        "message": "Webhook processed successfully.",
        "event": processed_event
    }), 200

if __name__ == "__main__":
    app.run(debug=True)

File: final_dataset_A/generated_code/chatgpt/java/ReceiveProcessWebhookDetailed-chatgpt.java
package com.example.demo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;
import java.util.LinkedHashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/webhook")
public class WebhookController {

    @PostMapping
    public ResponseEntity<Map<String, Object>> receiveWebhook(
            @RequestBody Map<String, Object> payload) {

        // Parse the incoming JSON payload
        String eventType = (String) payload.getOrDefault("eventType", "UNKNOWN");

        // Process the event
        System.out.println("Processing event: " + eventType);

        // Record the processed event
        Map<String, Object> eventRecord = new LinkedHashMap<>();
        eventRecord.put("eventType", eventType);
        eventRecord.put("processedAt", LocalDateTime.now());
        eventRecord.put("status", "PROCESSED");

        // Confirmation response
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("success", true);
        response.put("message", "Webhook processed successfully.");
        response.put("event", eventRecord);

        return ResponseEntity.ok(response);
    }
}